lib/config.js

```diff
@@ -150,79 +169,130 @@ const replaceConfigSlashes = exports.replaceConfigSlashes = config => {
 
 const normalizeChecker = anymatch;
 
+const checkFilesKeys = configFiles => {
+  const allowedFileTypes = ['javascripts', 'stylesheets', 'templates'];
+  const types = Object.keys(configFiles);
+
+  types.filter(type => allowedFileTypes.indexOf(type) === -1).forEach(type => {
+    logger.warn(`'${type}' is not an allowed 'files' key. You can only specify 'javascripts', 'stylesheets' or 'templates'.`);
+  });
+};
+
+const normalizeJoinConfig = joinTo => {
+  /* Can be used in `reduce` as `array.reduce(listToObj, {})`. */
+  const listToObj = (acc, elem) => {
+    acc[elem[0]] = elem[1];
+    return acc;
+  };
+
+  let object;
+  if (typeof joinTo === 'string') {
+    object = {};
+    object[joinTo] = /.+/;
+  } else {
+    object = joinTo;
+  }
+  const makeChecker = generatedFilePath => {
+    return [generatedFilePath, normalizeChecker(object[generatedFilePath])];
+  };
+  const subCfg = Object.keys(object).map(makeChecker).reduce(listToObj, {});
+  return subCfg;
+};
+
+const normalizePluginHelpers = (items, subCfg) => {
+  const vendorRe = /vendor/i;
+  if (!subCfg) return;
+  if (!items) {
+    items = (() => {
+      const destFiles = Object.keys(subCfg);
+      const joinMatch = destFiles.filter(file => {
+        return subCfg[file]('vendor/.');
```

Comment:
it's just the existing code moved into a separate function. to not introduce any breaking changes in this (probably not oft used) part, i think it very well makes sense to keep it as is